src/Model/Command/Interaction.php

```diff
@@ -34,12 +36,12 @@ private function __construct(
         string $context = 'default',
         int $timestamp = null
     ) {
-        $this->interactionType = $interactionType; // TODO: assert one of INTERACTION_TYPE_*
-        $this->userId = $userId; // TODO: assert format
-        $this->itemId = $itemId; // TODO: assert format
-        $this->value = $value; // TODO: assert value between 0-1
-        $this->context = $context; // TODO: assert format
-        $this->timestamp = $timestamp ?: time(); // TODO: assert format
+        $this->setInteractionType($interactionType);
+        $this->setUserId($userId);
+        $this->setItemId($itemId);
+        $this->setValue($value);
+        $this->setContext($context);
+        $this->setTimestamp($timestamp ?: time());
```

Comment:
Why not use `??` instead of `?:`

Reply:
`??` is useful when the value may not be set. But the `$timestamp` is always set (but may be null).

Reply:
`??` is also useful when the value might be `null`, which is the default value here. With `?:` both `null` and `0` gets coalesced to `time()`. I just wasn't sure if that was inteded.

Reply:
I will change the condition to `$timestamp === null`  to clearly state this covers the case when no value is passed (ie. default value should be used).